### PR TITLE
Add failing test fixture for AnnotationToAttributeRector

### PR DESCRIPTION
**Pull request by stephanvierkant**
# Failing Test for AnnotationToAttributeRector

Based on https://getrector.org/demo/f345dae3-9a60-4fc3-b838-d2f21ddbccf4

### CODE REVIEW

**Comment by stephanvierkant** [comment]
This wasn't expected. The demo removes arguments from the annotation/attribute and that's exactly what's going wrong when using Rector in my project.

How do I get the same result in the tests?

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/apiplatform.php.inc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * @ApiPlatform\Core\Annotation\ApiResource(
+ *     routePrefix="/demo/",
+ * )
+ */
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/apiplatform.php.inc:9]: this seems should be before class, not before namespace.

**Comment by TomasVotruba** [comment]
Indeed. The annotatoin must be in the test config, otherwise it won't be changed from annotation to api.

Also, could you re-use existing annotation? The bug is not in the class name, but in the content.

**Comment by stephanvierkant** [comment]
Interesting. I used the 'Create a test' button on the demo site.

Now fixed and the test is failing exactly as expected.

**Comment by TomasVotruba** [comment]
Thanks :+1: 

I think the demo create-test button does has a bug when it comes to putting namespace above docblock.

**Comment by TomasVotruba** [comment]
Here is directory with fixtures for making demo link:
https://github.com/rectorphp/getrector.org/blob/main/packages-tests/GitHubMagicLink/BodyFactory/FixtureBodyFactory/Fixture/expected_fixture_body.txt

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/apiplatform.php.inc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;
+
+/**
+ * @ApiPlatform\Core\Annotation\ApiResource(
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/apiplatform.php.inc:6]: it needs to be `\ApiPlatform` prefix as not in use statement as inside namespace.

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/apiplatform.php.inc:6]: It needs a use statement actually: 

```php
use ApiPlatform\Core\Annotation\ApiResource;

/**
 * @ApiResource(
 *     routePrefix="/demo/"
 * )
 */
 ```

**Comment by samsonasik** [comment]
@stephanvierkant I tried to update your fixture with the following:

```php
<?php

namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;

use ApiPlatform\Core\Annotation\ApiResource;

/**
 * @ApiResource(
 *     routePrefix="/demo/"
 * )
 */
final class DemoFile
{
}
?>
-----
<?php

namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture;

use ApiPlatform\Core\Annotation\ApiResource;

#[\ApiPlatform\Core\Annotation\ApiResource(routePrefix: '/demo/')]
final class DemoFile
{
}
?>
```

and the test is running ok after rebase against latest main:

Screen Shot 2021-06-22 at 19 20 18

**Comment by samsonasik** [comment]
It seems the issue happen on non-namespaced class:

Screen Shot 2021-06-22 at 19 31 19

**Comment by samsonasik** [comment]
@stephanvierkant I cherry-picked your commits at PR https://github.com/rectorphp/rector-src/pull/268

**Pull request closed without merging**